### PR TITLE
feat(test): Shared CRUD scenario

### PR DESCRIPTION
**Pull request by Cobalt0s**
# Problem

Manually copying and tweaking test scenarios for `create → search → update → validate → delete` is repetitive, clunky and error-prone.

# Fix
Introduced a high-level utility `testscenario` package to streamline common CRUD test flows.
As a demo, I refactored DynamicsCRM tests—now much shorter. This will also benefit other connectors like Insightly that follow the same write-delete pattern. Several existing have the same pattern, but I am not refactoring them until we agree on this change.

# Test
## DynamicsCRM
image

### CODE REVIEW

**Comment by Cobalt0s** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1578?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1579** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1579?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1578** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1578?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1578?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1577** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1577?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1576** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1576?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1575** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1575?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1560** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1560?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: test/dynamicscrm/write-delete/leads/main.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+type LeadCreatePayload struct {
+	LastName    string `json:"lastname,omitempty"`
+	FirstName   string `json:"firstname,omitempty"`
+	CompanyName string `json:"companyname,omitempty"`
+	Subject     string `json:"subject,omitempty"`
+}
+
+type LeadUploadPayload struct {
```
Comment (jlimatampersand) [inline test/dynamicscrm/write-delete/leads/main.go:22]: shouldn't this be `LeadUpdatePayload`?

**Review by jlimatampersand — APPROVED** [review]
I like this approach this is great. I left a small comment. Feel free to merge after resolving it. 

But I have 1 question to ask regarding this approach. This kind of assumes that CRUD is all available when writing test. But we may make PRs with just create, just update, or just delete. Bun until we have all 3, we don't have access to the test suite. How would you handle this?

File: test/dynamicscrm/write-delete/leads/main.go (hunk 1 above)
Comment (Cobalt0s) [reply test/dynamicscrm/write-delete/leads/main.go:22]: Yes! My bad.

**Comment by Cobalt0s** [comment]
> I like this approach this is great. I left a small comment. Feel free to merge after resolving it.
> 
> But I have 1 question to ask regarding this approach. This kind of assumes that CRUD is all available when writing test. But we may make PRs with just create, just update, or just delete. Bun until we have all 3, we don't have access to the test suite. How would you handle this?

Create and Update are typically implemented together. Adding Delete, where applicable, is usually straightforward, the easiest. So in most providers, the test scenario would be availble and fits perfectly to demonstrate implementation targeting Create/Upadte/Delete. 

So far I've added 10 connector tests with exact "template", they can be refactored. I am sure many more will act the same.

However, I agree this approach may not work well for connectors that lack proper Delete support—or in cases where only Create and Delete exist **without** Update. In such cases, we should consider creating an alternative test scenario. I foresee only a handful of these variants being needed to cover most edge cases.
